File: app/src/main/java/br/pedroso/upcomingmovies/movieslist/MoviesListActivity.java
package br.pedroso.upcomingmovies.movieslist;

import android.content.Intent;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import java.util.List;

import javax.inject.Inject;

import br.pedroso.upcomingmovies.MoviesApplication;
import br.pedroso.upcomingmovies.databinding.ActivityMoviesBinding;
import br.pedroso.upcomingmovies.di.ApplicationComponent;
import br.pedroso.upcomingmovies.domain.Movie;
import br.pedroso.upcomingmovies.moviedetails.ui.MovieDetailsActivity;
import br.pedroso.upcomingmovies.movieslist.adapter.MoviesAdapter;
import br.pedroso.upcomingmovies.movieslist.di.DaggerMoviesComponent;
import br.pedroso.upcomingmovies.movieslist.di.MoviesPresenterModule;

public class MoviesListActivity extends AppCompatActivity implements MoviesListView {

    private MoviesAdapter moviesAdapter;

    private ActivityMoviesBinding binding;

    @Inject
    MoviesListPresenter presenter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        injectPresenter();
        setupView();
    }

    @Override
    protected void onResume() {
        super.onResume();
        presenter.resume();
    }

    @Override
    protected void onPause() {
        super.onPause();
        presenter.pause();
    }

    private void setupView() {
        binding = ActivityMoviesBinding.inflate(getLayoutInflater());

        setContentView(binding.getRoot());

        setupRecyclerViewMovies();
    }

    private void setupRecyclerViewMovies() {
        moviesAdapter = new MoviesAdapter(presenter::onMovieClick);
        binding.recyclerViewMovies.setAdapter(moviesAdapter);
    }

    private void injectPresenter() {
        ApplicationComponent applicationComponent = ((MoviesApplication) getApplication()).getApplicationComponent();

        DaggerMoviesComponent.builder()
                .applicationComponent(applicationComponent)
                .moviesPresenterModule(new MoviesPresenterModule(this))
                .build()
                .inject(this);
    }

    @Override
    public void renderMoviesList(List<Movie> moviesList) {
        moviesAdapter.updateAdapterData(moviesList);
    }

    @Override
    public void startMovieDetailsActivity(Integer movieId) {
        Intent intent = new Intent(this, MovieDetailsActivity.class);
        intent.putExtra(MovieDetailsActivity.EXTRA_MOVIE_ID, movieId);
        startActivity(intent);
    }

    @Override
    public void cleanMoviesList() {
        moviesAdapter.clearItems();
    }
}

File: app/src/main/kotlin/br/pedroso/upcomingmovies/movieslist/MoviesListActivity.kt
package br.pedroso.upcomingmovies.movieslist

import android.os.Bundle
import androidx.activity.compose.setContent
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.ui.Modifier
import br.pedroso.upcomingmovies.designsystem.theme.UpcomingMoviesTheme
import br.pedroso.upcomingmovies.moviedetails.MovieDetailsActivity
import dagger.hilt.android.AndroidEntryPoint

@AndroidEntryPoint
class MoviesListActivity : AppCompatActivity() {
    private val viewModel: MoviesListViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContent {
            UpcomingMoviesTheme {
                MoviesListScreen(
                    modifier = Modifier.fillMaxSize(),
                    viewModel = viewModel,
                    navigateToMovieDetails = { movieId ->
                        MovieDetailsActivity.openMovieDetails(context = this, movieId = movieId)
                    }
                )
            }
        }
    }
}
